### PR TITLE
feat(api): clarify the notification steps creation

### PR DESCRIPTION
**Pull request by p-fernandez**
### What change does this PR introduce?

Small refactor to aggregate the logic accordingly.

### Why was this change needed?

Inside of the use case that was in charge on deciding if a digest step should be created or not we were doing more things, like filtering the active steps for the notification, adding extra complexity to that use case.
These changes should facilitate the vision if we are committing to decisions on creating or not digest steps before we should.

### CODE REVIEW

File: apps/api/src/app/message-template/usecases/update-message-template/update-message-template.usecase.ts [hunk 1]
```diff
@@ -110,23 +110,25 @@ export class UpdateMessageTemplate {
       );
     }
 
-    const changeId = await this.changeRepository.getChangeId(
-      command.environmentId,
-      ChangeEntityTypeEnum.MESSAGE_TEMPLATE,
-      item._id
-    );
+    if (item._id) {
```
Comment (p-fernandez) [inline apps/api/src/app/message-template/usecases/update-message-template/update-message-template.usecase.ts:113]: Change to comply with setting _id as optional for MessageTemplateEntity.

File: apps/api/src/app/message-template/usecases/create-message-template/create-message-template.usecase.ts [hunk 2]
```diff
@@ -37,7 +37,10 @@ export class CreateMessageTemplate {
       actor: command.actor,
     });
 
-    item = (await this.messageTemplateRepository.findById(item._id)) as MessageTemplateEntity;
+    if (item?._id) {
```
Comment (p-fernandez) [inline apps/api/src/app/message-template/usecases/create-message-template/create-message-template.usecase.ts:40]: Change to comply with setting _id as optional for MessageTemplateEntity.

**Review by LetItRock — APPROVED** [review]
looks good to me 🥇

File: apps/api/src/app/events/usecases/digest-filter-steps/digest-filter-steps.usecase.ts [hunk 3]
```diff
@@ -23,44 +25,35 @@ export class DigestFilterSteps {
...
     return {
       template: {
         _environmentId: command.environmentId,
         _organizationId: command.organizationId,
         _creatorId: command.userId,
+        _layoutId: null,
         type: StepTypeEnum.TRIGGER,
         content: '',
-      } as MessageTemplateEntity,
+      },
```
Comment (djabarovgeorge) [inline apps/api/src/app/events/usecases/digest-filter-steps/digest-filter-steps.usecase.ts:56]: Not relevant to this PR but are you aware of why _layoutId is nullable and not optional?

**Pull request merged**